### PR TITLE
Test dynamic surface switch

### PR DESCRIPTION
**Pull request by blasten**
## Description

This PR tests that the code paths that switches surfaces dynamically based on whether there are Android views in the current frame.

## Related Issues

https://github.com/flutter/flutter/issues/61119

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/integration_tests/hybrid_android_views/android/app/src/main/java/io/flutter/integration/androidviews/MainActivity.java [hunk 1]
```diff
@@ -36,6 +41,72 @@ private View getFlutterView() {
...
+        if (view instanceof FlutterView) {
+            return "FlutterView";
+        }
+        if (view instanceof ViewGroup) {
+            return "ViewGroup";
+        }
+        return "View";
+    }
+
+    private void getViewHierarchy(View current, String padding, StringBuilder builder) {
```
Comment (cyanglaz) [inline dev/integration_tests/hybrid_android_views/android/app/src/main/java/io/flutter/integration/androidviews/MainActivity.java:63]: As a recursive void method, the name `getViewHierarchy` seems confusing. How about `updateViewHierarchyString`?

File: dev/integration_tests/hybrid_android_views/lib/future_data_handler.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    if (_handlers[message] == null) {
+      return 'Unsupported driver message: $message.\n'
+             'Supported messages are: ${_handlers.keys}.';
+    }
+    final Function() handler = await _handlers[message].future;
+    return handler();
+  }
+}
+
+FutureDataHandler driverDataHandler = FutureDataHandler();
```
Comment (cyanglaz) [inline dev/integration_tests/hybrid_android_views/lib/future_data_handler.dart:33]: Line at the end of file

**Review by cyanglaz — APPROVED** [review]
LGTM modulo nits

File: dev/integration_tests/hybrid_android_views/lib/future_data_handler.dart (hunk 2 above)
Comment (blasten) [reply dev/integration_tests/hybrid_android_views/lib/future_data_handler.dart]: Done.

File: dev/integration_tests/hybrid_android_views/android/app/src/main/java/io/flutter/integration/androidviews/MainActivity.java (hunk 1 above)
Comment (blasten) [reply dev/integration_tests/hybrid_android_views/android/app/src/main/java/io/flutter/integration/androidviews/MainActivity.java]: Done.

**Pull request merged**